### ISSUE TITLE
Add support for anonymous pipes

### ISSUE THREAD

**Issue by notgull** [smol-rs/polling#82]
It would be useful if this crate supported Windows anonymous pipes.

I probably can't make any progress on this until #76 is complete, as this would require the backend to not be `wepoll`.

**Comment by kchibisov** [comment]
Given that your patch merged, how far are we from anon pipes? Asking since rust time bombed mio 0.6.x (it has future incompatible code) and we must find an alternative.

**Comment by notgull** [comment]
We still need to support direct `OVERLAPPED` operations in our use case. However, looking at your use case, it looks like you use [`mio-anonymous-pipes`], which "cheats" by putting the pipe onto a thread pool. Therefore, you might be able to get away with just wrapping the pipe in an [`Unblock`](https://docs.rs/blocking/latest/blocking/struct.Unblock.html) and using it like that.

**Comment by notgull** [comment]
Not as much of a priority anymore as we've worked around this limitation in `alacritty`. This is mostly a duplicate of #97 now, as we would need overlapped operation support to truly support anonymous pipes in `polling`.
